### PR TITLE
clear MPNumbers before return

### PR DESCRIPTION
**Pull request by mbkma**

### CODE REVIEW

File: src/mp-trigonometric.c [hunk 1]
```diff
@@ -119,6 +119,7 @@ mp_tan(const MPNumber *x, MPAngleUnit unit, MPNumber *z)
         /* Translators: Error displayed when tangent value is undefined */
         mperr(_("Tangent is undefined for angles that are multiples of π (180°) from π∕2 (90°)"));
         mp_set_from_integer(0, z);
+        mp_clear(&x_radians); mp_clear(&pi); mp_clear(&t1);
```
Comment (rbuj) [inline src/mp-trigonometric.c]: One sentence per line.

File: src/mp-trigonometric.c [hunk 2]
```diff
@@ -146,6 +147,7 @@ mp_asin(const MPNumber *x, MPAngleUnit unit, MPNumber *z)
         /* Translators: Error displayed when inverse sine value is undefined */
         mperr(_("Inverse sine is undefined for values outside [-1, 1]"));
         mp_set_from_integer(0, z);
+        mp_clear(&x_max); mp_clear(&x_min);
```
Comment (rbuj) [inline src/mp-trigonometric.c]: One sentence per line.

File: src/mp-trigonometric.c [hunk 3]
```diff
@@ -169,6 +171,7 @@ mp_acos(const MPNumber *x, MPAngleUnit unit, MPNumber *z)
         /* Translators: Error displayed when inverse sine value is undefined */
         mperr(_("Inverse cosine is undefined for values outside [-1, 1]"));
         mp_set_from_integer(0, z);
+        mp_clear(&x_max); mp_clear(&x_min);
```
Comment (rbuj) [inline src/mp-trigonometric.c]: One sentence per line.

File: src/mp-trigonometric.c [hunk 4]
```diff
@@ -193,6 +196,7 @@ mp_atan(const MPNumber *x, MPAngleUnit unit, MPNumber *z)
         /* Translators: Error displayed when inverse sine value is undefined */
         mperr(_("Arctangent function is undefined for values i and -i"));
         mp_set_from_integer(0, z);
+        mp_clear(&i); mp_clear(&minus_i);
```
Comment (rbuj) [inline src/mp-trigonometric.c]: One sentence per line.

File: src/mp-trigonometric.c [hunk 5]
```diff
@@ -264,6 +269,7 @@ mp_atanh(const MPNumber *x, MPNumber *z)
          /* Translators: Error displayed when inverse hyperbolic tangent value is undefined */
         mperr(_("Inverse hyperbolic tangent is undefined for values outside (-1, 1)"));
         mp_set_from_integer(0, z);
+        mp_clear(&x_max); mp_clear(&x_min);
```
Comment (rbuj) [inline src/mp-trigonometric.c]: One sentence per line.

**Comment by rbuj** [comment]
You can use labels, for instance:

```C
void
mp_tan(const MPNumber *x, MPAngleUnit unit, MPNumber *z)
{
    MPNumber x_radians = mp_new();
    MPNumber pi = mp_new();
    MPNumber t1 = mp_new();

    convert_to_radians(x, unit, &x_radians);
    mp_get_pi(&pi);
    mp_divide_integer(&pi, 2, &t1);
    mp_subtract(&x_radians, &t1, &t1);
    mp_divide(&t1, &pi, &t1);    

    if (mp_is_integer(&t1)) {
        /* Translators: Error displayed when tangent value is undefined */
        mperr(_("Tangent is undefined for angles that are multiples of π (180°) from π∕2 (90°)"));
        mp_set_from_integer(0, z);
        goto mp_tan_cleanup;
    }

    if (mp_is_complex(x))
        mp_set_from_mp(x, z);
    else
        mp_set_from_mp(&x_radians, z);
    mpc_tan(z->num, z->num, MPC_RNDNN);

mp_tan_cleanup:
    mp_clear(&x_radians);
    mp_clear(&pi);
    mp_clear(&t1);
}
```

**Comment by mbkma** [comment]
@rbuj do you agree with this solution?
```c
void
mp_clear_n(int arg_count, ...)
{
    MPNumber *x;
    va_list args;

    va_start(args, arg_count);
    for (int i = 0; i < arg_count ; i++)
    {
        x = va_arg(args, MPNumber*);
        mp_clear(x);
    }
    va_end(args);
}
```

```c
void
mp_tan(const MPNumber *x, MPAngleUnit unit, MPNumber *z)
{
    MPNumber x_radians = mp_new();
    MPNumber pi = mp_new();
    MPNumber t1 = mp_new();

    convert_to_radians(x, unit, &x_radians);
    mp_get_pi(&pi);
    mp_divide_integer(&pi, 2, &t1);
    mp_subtract(&x_radians, &t1, &t1);
    mp_divide(&t1, &pi, &t1);    

    if (mp_is_integer(&t1)) {
        /* Translators: Error displayed when tangent value is undefined */
        mperr(_("Tangent is undefined for angles that are multiples of π (180°) from π∕2 (90°)"));
        mp_set_from_integer(0, z);
        mp_clear_n(3, &x_radians, &pi, &t1);
        return;
    }

    if (mp_is_complex(x))
        mp_set_from_mp(x, z);
    else
        mp_set_from_mp(&x_radians, z);
    mpc_tan(z->num, z->num, MPC_RNDNN);
    mp_clear_n(3, &x_radians, &pi, &t1);
}
```

**Comment by rbuj** [comment]
Do you want to use varargs for destroying all variabes in one line? Why not use varargs to create them? Please, keep the code as clear as possible and don't complicate it if you don't need to.

**Comment by rbuj** [comment]
Test:
```
$ CFLAGS="-ggdb3 -O0 -fsanitize=address" ./autogen.sh --prefix=/usr && make
$ ./src/test-mp-equation
```
Before (master):
```
SUMMARY: AddressSanitizer: 213285 byte(s) leaked in 2884 allocation(s).
```
After (PR+git rebase master):
```
SUMMARY: AddressSanitizer: 212469 byte(s) leaked in 2878 allocation(s).
```

**Comment by mbkma** [comment]
yes, you are right. Thanks for reviewing!

**Pull request merged**